breadbox/breadbox/service/metadata.py

```diff
@@ -36,6 +42,82 @@ def get_tabular_dataset_metadata_annotations(
     return filtered_metadata_vals
 
 
+def get_matrix_dataset_feature_metadata(
+    db: SessionWithUser, dataset: MatrixDataset, metadata_col_name: str,
+) -> dict[str, Any]:
+    """
+    For the given matrix dataset, load a column from the associated metadata.
+    The result will only include given ids which exist in both the dataset and the metadata.
+    """
+    full_metadata_col = types_crud.get_dimension_type_metadata_col(
+        db, dimension_type_name=dataset.feature_type_name, col_name=metadata_col_name
+    )
+    # Filter the metadata to only include the given IDs belonging to this dataset
+    dataset_features = dataset_crud.get_dataset_features(db, dataset)
+    filtered_metadata_vals = {}
+    for feature in dataset_features:
+        metadata_val = full_metadata_col.get(feature.given_id)
+        if metadata_val is not None:
+            filtered_metadata_vals[feature.given_id] = metadata_val
+    return filtered_metadata_vals
+
+
+def get_matrix_dataset_sample_metadata(
+    db: SessionWithUser, dataset: MatrixDataset, metadata_col_name: str,
+) -> dict[str, Any]:
+    """
+    For the given matrix dataset, load a column from the associated metadata.
+    The result will only include given ids which exist in both the dataset and the metadata.
+    """
+    full_metadata_col = types_crud.get_dimension_type_metadata_col(
+        db, dimension_type_name=dataset.sample_type_name, col_name=metadata_col_name
+    )
+    # Filter the metadata to only include the given IDs belonging to this dataset
+    dataset_samples = dataset_crud.get_dataset_samples(db, dataset)
+    filtered_metadata_vals = {}
+    for sample in dataset_samples:
+        metadata_val = full_metadata_col.get(sample.given_id)
+        if metadata_val is not None:
+            filtered_metadata_vals[sample.given_id] = metadata_val
+    return filtered_metadata_vals
+
+
+def get_dataset_feature_labels_by_id(
+    db: SessionWithUser, user: str, dataset: MatrixDataset,
+) -> dict[str, str]:
+    """
+    Try loading feature labels from metadata.
+    If there are no labels in the metadata or there is no metadata, then just return the feature names.
+    """
+    if dataset.feature_type_name is not None:
+        metadata_labels_by_given_id = get_matrix_dataset_feature_metadata(
+            db=db, dataset=dataset, metadata_col_name="label"
+        )
+        if metadata_labels_by_given_id:
+            return metadata_labels_by_given_id
+
+    # If there are no labels or there is no feature type, return the given IDs
+    all_dataset_features = dataset_crud.get_dataset_features(db, dataset)
+    return {feature.given_id: feature.given_id for feature in all_dataset_features}
+
+
+def get_dataset_sample_labels_by_id(
```

Comment:
I like how we are being explicit with what type of dataset in this method's parameter. However, it would also be nice for the name of the method to be more explicit too like `get_matrix_dataset_sample_labels_by_id` similar to the below method `get_tabular_dataset_labels_by_id`